tests/test_streaming.py

```diff
@@ -272,48 +278,55 @@ async def test_respawn_consumer_task(
 
     async with tractor.open_nursery() as n:
 
-        stream = await(await n.run_in_actor(
+        portal = await n.start_actor(
+            name='streamer',
+            enable_modules=[__name__]
+        )
+        async with portal.open_stream_from(
             stream_data,
             seed=11,
-            name='streamer',
-        )).result()
+        ) as stream:
+
+            expect = set(range(11))
+            received = []
 
-        expect = set(range(11))
-        received = []
+            # this is the re-spawn task routine
+            async def consume(task_status=trio.TASK_STATUS_IGNORED):
+                print('starting consume task..')
+                nonlocal stream
 
-        # this is the re-spawn task routine
-        async def consume(task_status=trio.TASK_STATUS_IGNORED):
-            print('starting consume task..')
-            nonlocal stream
+                with trio.CancelScope() as cs:
+                    task_status.started(cs)
 
-            with trio.CancelScope() as cs:
-                task_status.started(cs)
+                    # shield stream's underlying channel from cancellation
+                    with stream.shield():
 
-                # shield stream's underlying channel from cancellation
-                with stream.shield():
+                        async for v in stream:
+                            print(f'from stream: {v}')
+                            expect.remove(v)
+                            received.append(v)
 
-                    async for v in stream:
-                        print(f'from stream: {v}')
-                        expect.remove(v)
-                        received.append(v)
+                    print('exited consume')
 
-                print('exited consume')
+            async with trio.open_nursery() as ln:
+                cs = await ln.start(consume)
 
-        async with trio.open_nursery() as ln:
-            cs = await ln.start(consume)
+                while True:
 
-            while True:
+                    await trio.sleep(0.1)
 
-                await trio.sleep(0.1)
+                    if received[-1] % 2 == 0:
 
-                if received[-1] % 2 == 0:
+                        print('cancelling consume task..')
+                        cs.cancel()
 
-                    print('cancelling consume task..')
-                    cs.cancel()
+                        # respawn
+                        cs = await ln.start(consume)
 
-                    # respawn
-                    cs = await ln.start(consume)
+                    if not expect:
+                        print("all values streamed, BREAKING")
+                        break
 
-                if not expect:
-                    print("all values streamed, BREAKING")
-                    break
+        # TODO: this is justification for a
```

Comment:
Ahh so one question is if we want to move forward with a `ActorNursery.run_in_actor()` equivalent for streams.  Pretty sure we go over this in https://github.com/goodboy/tractor/issues/172.